### DISCUSSION TITLE
Support XMP sidecars

### DISCUSSION THREAD

**Discussion by StefanOltmann** [ente-io/ente#4046] [Enhancements]
Support XMP sidecar files for RAW images, and provide users with the option to choose whether changes should be embedded directly into JPG and PNG files or written to a sidecar file.

Regarding exports, I noticed you use a custom JSON format. Could you replace it with standard XMP sidecar files instead?

**Comment by annargrs** [comment]
At least *reading* the metadata from sidecar files is a must. Otherwise there is no way to add metadata for raw files into ente.

**Reply by StefanOltmann** [reply]
Completely agree, but I doubt they'll take that step. I've asked about it several times, and it seems metadata handling just isn't a priority for them. Ente is more of a gallery app than a true photo organizer.

**Comment by LittleBlueBreeze** [comment]
Yes please, this is so needed. Can't really get out of iCloud Photos as long as my XMP data is unable to be imported into ente.

**Reply by StefanOltmann** [reply]
iCloud data has XMP?

**Reply by LittleBlueBreeze** [reply]
Apparently, yes. I had to use a command line tool called osxphotos to write the XMP sidecar data to the metadata of the photos/videos themselves. Very tricky process.

**Reply by scrapp08** [reply]
> Apparently, yes. I had to use a command line tool called osxphotos to write the XMP sidecar data to the metadata of the photos/videos themselves. Very tricky process.

How did you do it? I think i'm running into the same issue, i've got a bunch of photos all with the completely wrong metadata.

**Reply by StefanOltmann** [reply]
I guess this is the tool mentioned: https://github.com/RhetTbull/osxphotos

**Reply by scrapp08** [reply]
Yeah, curious how they wrote the XMP data to the media itself though. There's some metadata commands, but looks as if you might have to recurse through each photo and add the metadata, which does indeed sound like a pain.

**Reply by StefanOltmann** [reply]
Indeed. LittleBlueBreeze may have a script to share with us. ;)

**Reply by LittleBlueBreeze** [reply]
These are the steps I used! https://www.reddit.com/r/enteio/comments/1is6zs9/importing_from_icloud_photos/
By the way, not everyone on Github is a "he" lol :p

**Reply by StefanOltmann** [reply]
> By the way, not everyone on Github is a "he" lol :p

😅

I guess I avoid using 'they' for an unknown gender because, to me, it refers to a group. 😄

I fixed my post. ;)

**Reply by LittleBlueBreeze** [reply]
Thanks! hehe

**Reply by scrapp08** [reply]
Have amended my comment as well, apologies. I'll give it a try later, and will let you know how it goes.

**Reply by LittleBlueBreeze** [reply]
Good luck!!

**Reply by scrapp08** [reply]
Did a dry run, no errors. Did a proper run, errors. Most are yellow saying "Invalid entry from MakerNotes", some reds though saying "Not a valid HEIC/PNG, looks more like a JPEG", or "Error reading OtherImageStart data"

**Reply by StefanOltmann** [reply]
You can drop your files into https://stefan-oltmann.de/exif-viewer to take a closer look.

**Reply by LittleBlueBreeze** [reply]
@scrapp08 Ahh that is frustrating, are you able to figure out which photos exactly are giving you the "red" errors? I'm just shooting from the hip here, but for the "yellow" errors, maybe you can drop them all into a folder and use ExifTool to change the MakerNotes

**Reply by StefanOltmann** [reply]
One of the great features of ExifTool is that it also repairs minor errors, so it’s likely that just resaving them helps already.

My EXIF Viewer is highly inspired by ExifTool HtmlDump, but it doesn’t mark errors like wrong offsets. So in doubt look at the original HtmlDump.
